components/ai_chat/resources/page/components/conversation_list/index.tsx

```diff
@@ -100,8 +105,12 @@ function ConversationList() {
                 <div className={avatarStyles}>
                   <Icon name={isHuman ? 'user-circle' : 'product-brave-leo'} />
                 </div>
-                <div className={styles.message}>
-                  {turn.text}
+                <div
+                  className={styles.message}
+                >
+                  <div dangerouslySetInnerHTML={{ __html: escapeHTMLPolicy.createHTML(turn.text
```

Comment:
Can we get around the need to use innerHTML. One option would be to use your regex here to build an array of matches and then render everything in a loop
- regular text block in between a match
- code block text wrapped with special element or class

And for each regular block you could use the same pattern for any inline code block matches.

function MessageTextPart ({messageText}) {
   const {beforeCodeBlock, duringCod
}

Reply:
ok, changed. i'm building the string into parts and wrapping relevant html around it. this works, nicely. 

however, i'm curious how expensive using regex is here